File: 1441.build-an-array-with-stack-operations.cpp
/*
 * @lc app=leetcode id=1441 lang=cpp
 *
 * [1441] Build an Array With Stack Operations
 */
#include <string>
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<string> buildArray(vector<int> &target, int n)
    {
        int value = 1;
        vector<string> res;
        for (auto n : target)
        {
            while (value < n)
            {
                res.push_back("Push");
                res.push_back("Pop");
                ++value;
            }
            res.push_back("Push");
            ++value;
        }
        return res;
    }
};
// @lc code=end

File: 1441.build-an-array-with-stack-operations.py
#
# @lc app=leetcode id=1441 lang=python3
#
# [1441] Build an Array With Stack Operations
#

# @lc code=start
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        value = 1
        res = []
        for n in target:
            res.extend(["Push", "Pop"]*(n-value))
            res.append("Push")
            value = n+1
        return res
# @lc code=end
